### PR TITLE
add log with logrus lib

### PR DESCRIPTION
**Pull request by SamYuan1990**
add log with official lib
add more block for testing to see log

Signed-off-by: SamYuan1990 <yy19902439@126.com>

### CODE REVIEW

**Comment by guoger** [comment]
`log` in golang stdlib is **rarely** used in production. Instead, we could switch to some more popular libs, like [Logrus](https://github.com/sirupsen/logrus) or [glog](https://github.com/golang/glog)

**Comment by SamYuan1990** [comment]
> `log` in golang stdlib is **rarely** used in production. Instead, we could switch to some more popular libs, like [Logrus](https://github.com/sirupsen/logrus) or [glog](https://github.com/golang/glog)
> 
> The point is that we should be able to get **leveled** logs (fatal, warn, info, debug)

ok, will update this pr later.

**Comment by SamYuan1990** [comment]
it seems that https://github.com/sirupsen/logrus log format is string, so use `fmt.Sprint` refactor some logging lines.
https://github.com/sirupsen/logrus/blob/master/entry.go#L217

File: infra/observer.go [hunk 1]
```diff
@@ -4,16 +4,18 @@ import (
 	"fmt"
 	"time"
 
+	log "github.com/sirupsen/logrus"
+
```
Comment (guoger) [inline infra/observer.go:8]: remove empty line

File: infra/observer.go [hunk 2]
```diff
@@ -48,14 +50,14 @@ func (o *Observer) Start(N int, now time.Time) {
 
 		fb := r.Type.(*peer.DeliverResponse_FilteredBlock)
 		n = n + len(fb.FilteredBlock.FilteredTransactions)
-		fmt.Printf("Time %8.2fs\tBlock %6d\tTx %6d\n", time.Since(now).Seconds(), fb.FilteredBlock.Number, len(fb.FilteredBlock.FilteredTransactions))
+		o.logger.Info("Time " + fmt.Sprint(time.Since(now).Seconds()) + "s Block " + fmt.Sprint(fb.FilteredBlock.Number) + " Tx " + fmt.Sprint(len(fb.FilteredBlock.FilteredTransactions)))
```
Comment (guoger) [inline infra/observer.go:53]: `o.logger.Infof` so you don't need `fmt.Sprintf`

File: infra/proposer.go [hunk 3]
```diff
@@ -5,6 +5,8 @@ import (
 	"fmt"
 	"io"
 
+	log "github.com/sirupsen/logrus"
+
```
Comment (guoger) [inline infra/proposer.go:9]: remove empty line

File: main.go [hunk 4]
```diff
@@ -6,12 +6,17 @@ import (
 	"strconv"
 	"time"
 
+	log "github.com/sirupsen/logrus"
+
```
Comment (guoger) [inline main.go:10]: empty line

File: main.go [hunk 5]
```diff
@@ -6,12 +6,17 @@ import (
...
 	"github.com/guoger/stupid/infra"
 )
 
 func main() {
+	log.SetOutput(os.Stdout)
+	log.SetLevel(log.WarnLevel)
+	logger := log.New()
 	if len(os.Args) != 3 {
-		fmt.Printf("Usage: stupid config.json 500\n")
+		logger.Panicln("Usage: stupid config.json 500")
```
Comment (guoger) [inline main.go:19]: We should keep this to use `fmt.Println`, since it's not exactly a log, but user instruction

File: main.go [hunk 6]
```diff
@@ -65,6 +70,6 @@ func main() {
 	duration := time.Since(start)
 	close(done)
 
-	fmt.Printf("tx: %d, duration: %+v, tps: %f\n", N, duration, float64(N)/duration.Seconds())
+	logger.Info("tx: " + fmt.Sprint(N) + " duration: " + fmt.Sprint(duration) + ", tps: " + fmt.Sprint(float64(N)/duration.Seconds()))
```
Comment (guoger) [inline main.go:73]: keep using `fmt.Print` here, since this is not log

File: infra/proposer.go (hunk 3 above)
Comment (SamYuan1990) [reply infra/proposer.go:9]: the empty line is added by vs code automatic.

File: main.go (hunk 4 above)
Comment (SamYuan1990) [reply main.go:10]: the empty line is added by vs code automatic.

File: infra/observer.go (hunk 1 above)
Comment (SamYuan1990) [reply infra/observer.go:8]: Auto added by vs code.

Comment (SamYuan1990) [reply infra/observer.go:8]: auto added by vs code

**Comment by SamYuan1990** [comment]
@guoger , not sure adding `export stupid_loglevel=debug` into integration test or not.

File: main.go [hunk 7]
```diff
@@ -7,11 +7,20 @@ import (
 	"time"
 
 	"github.com/guoger/stupid/infra"
+	log "github.com/sirupsen/logrus"
 )
 
+const loglevel = "stupid_loglevel"
```
Comment (guoger) [inline main.go:13]: by convention, env var is UPPERCASE

File: main.go [hunk 8]
```diff
@@ -7,11 +7,20 @@ import (
...
+
 func main() {
+	if lvl, err := log.ParseLevel(os.Getenv(loglevel)); err != nil {
+		log.SetLevel(log.WarnLevel)
+	} else {
+		log.SetLevel(lvl)
+	}
+	logger := log.New()
 	if len(os.Args) != 3 {
-		fmt.Printf("Usage: stupid config.json 500\n")
```
Comment (guoger) [inline main.go:14]: we still need `\n` at end (or switch to `Println`)

File: infra/logger.go [hunk 9]
```diff
@@ -0,0 +1,21 @@
+package infra
```
Comment (guoger) [inline infra/logger.go:1]: it's really not necessary to extract this into a function - it's simple, and only used once in `main`

File: infra/logger.go [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+import (
+	"os"
+
+	log "github.com/sirupsen/logrus"
+)
+
+const loglevel = "STUPID_LOGLEVEL"
+
+func NewLogger() *log.Logger {
+	log.SetLevel(log.InfoLevel)
```
Comment (guoger) [inline infra/logger.go:12]: `log.New` is not affected by `log.SetLevel`, which makes this line useless

File: infra/logger_test.go [hunk 11]
```diff
@@ -0,0 +1,34 @@
+package infra_test
```
Comment (guoger) [inline infra/logger_test.go:1]: it's generally good to have more comprehensive UTs, however i don't think this logger test is necessary. It's simply testing `logrus`, which is **not our pkg**

**Comment by SamYuan1990** [comment]
@guoger ,

the origin plan is for testing with environment variable.
if export STUPID_LOGLEVEL=\"debug\"
then no debug level log appears.
```
time="2020-06-23T11:24:09Z" level=info msg="Start sending transactions."
Time     0.26s	Block      8	Tx     10
Time     0.35s	Block      9	Tx     10
Time     0.48s	Block     10	Tx     10
Time     0.61s	Block     11	Tx     10
Time     0.64s	Block     12	Tx     10
Time     0.66s	Block     13	Tx     10
Time     0.67s	Block     14	Tx     10
Time     0.68s	Block     15	Tx     10
Time     0.69s	Block     16	Tx     10
Time     0.70s	Block     17	Tx     10
time="2020-06-23T11:24:10Z" level=info msg="Completed processing transactions."
tx: 100, duration: 698.814074ms, tps: 143.099579
```

**Comment by SamYuan1990** [comment]
either works for `export STUPID_LOGLEVEL=debug`

**Pull request merged**